test/jdk/java/nio/file/Files/InterruptCopy.java

```diff
@@ -75,19 +86,26 @@ static void doTest(Path dir) throws Exception {
         ScheduledExecutorService pool =
             Executors.newSingleThreadScheduledExecutor();
         try {
-            // copy source to target in main thread, interrupting it after a delay
+            // copy source to target in main thread, interrupting it
+            // at a fixed rate after a delay
             final Thread me = Thread.currentThread();
-            Future<?> wakeup = pool.schedule(new Runnable() {
+            Future<?> wakeup = pool.scheduleAtFixedRate(new Runnable() {
```

Comment:
Is it really a good idea? The interrupted state should stick to the thread until it is cleared. It should not really matter if you interrupt the thread too early - but if you interrupt the thread *again* after the File copy was interrupted, won't it skew the logic in the test? Especially WRT double checking that the interrupt status was/wasn't cleared? Also shouldn't you cancel this wakeup at some point before testing cancel below?

Reply:
I was not sure about the fixed rate firing here so you might be right. The wakeup is cancelled by `wakeup.get()` unless I misunderstand.

Reply:
Correction: I think an explicit `wakeup.cancel()` is needed.

Reply:
Apparently setting the interrupted state prior to the copy does _not_ cause the copy to be interrupted.